### ISSUE TITLE
Support sync with Google Cloud Source Repos

### ISSUE THREAD

**Issue by gingersnapz** [forseti-security/terraform-google-forseti#460]
Currently I have not found any supporting documents that describe syncing configuration policies with a Cloud Source repo instead of a Github or GCS bucket. 
There have been examples scattered that show manual ways to enable cloning via generated credentials for a user but not using a service account with proper permissions to a Cloud Source Repo. I have attempted to add the ssh key (generated by the terraform module when config validator is enabled) to the repo manually to allow cloning but Forseti config validator container still reports issues of access being denied. 
I'm using the recommended repo url format documented [here](https://cloud.google.com/source-repositories/docs/cloning-repositories): git clone ssh://[EMAIL]@source.developers.google.com:2022/p/[PROJECT_ID]/r/[REPO_NAME]

using the service account email.
Is there any specific configuration that is required to make these play nice? Any help is appreciated.

**Comment by red2k18** [comment]
@gingersnapz, thanks for reaching out! Yes, we have documented the [recommended practice](https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#provide-policies-to-forseti-server) i.e GitHub. Will discuss with the team the possibility of adding support for other practices such as Cloud Source repository and will update this issue. 

Thanks,
Rehana

**Comment by gkowalski-google** [comment]
@gingersnapz Can you provide any of the error messages that you get from the git-sync container?

**Comment by gingersnapz** [comment]
`main.go:299] "msg"="unexpected error syncing repo, will retry" "error"="error running command: exit status 128: \"Cloning into '/tmp/git'...\\nfatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.\\n\"" " `

I've added the SSH key that Forseti generates to the source repo and my repo url follows the format that I've commented above. I've attempted many different methods to get Forseti to sync using Google Cloud Source but have never gotten it to work. I've successfully used gcloud to clone the repo to the Forseti server instance but setting the repo url via terraform module and attempting to sync fails.
